openedx_tagging/core/tagging/import_export/actions.py

```diff
@@ -0,0 +1,444 @@
+            return (
+                taxonomy_tag.parent is not None
+                and taxonomy_tag.parent.external_id != tag.parent_id
+            ) or (taxonomy_tag.parent is None and tag.parent_id is not None)
+        except Tag.DoesNotExist:
+            return False
+
+    def validate(self, indexed_actions) -> List[ImportActionError]:
+        """
+        Validates the update parent action
+        """
+        errors = []
+
+        # Parent validation
+        if self.tag.parent_id:
+            error = self._validate_parent(indexed_actions)
+            if error:
+                errors.append(error)
+
+        return errors
+
+    def execute(self):
+        """
+        Updates the parent of a tag
+        """
+        tag = self._get_tag()
+        parent = None
+        if self.tag.parent_id:
+            parent = self.taxonomy.tag_set.get(external_id=self.tag.parent_id)
+        tag.parent = parent
+        tag.save()
+
+
+class RenameTag(ImportAction):
+    """
+    Action for rename a Tag
+
+    Action created if there is a change on the tag value
+
+    Validations:
+    - Value duplicates with tags on the database.
+    - Value duplicates with previous create and rename actions.
+    """
+
+    name = "rename"
+
+    def __str__(self):
+        taxonomy_tag = self._get_tag()
+        return str(
+            _(
+                f"Rename tag value of tag (id={taxonomy_tag.id}) "
+                f"from '{taxonomy_tag.value}' to '{self.tag.value}'"
+            )
+        )
+
+    @classmethod
+    def valid_for(cls, taxonomy: Taxonomy, tag) -> bool:
+        """
+        Validates if there is a change on the tag value
+        """
+        try:
+            taxonomy_tag = taxonomy.tag_set.get(external_id=tag.id)
+            return taxonomy_tag.value != tag.value
+        except Tag.DoesNotExist:
+            return False
+
+    def validate(self, indexed_actions) -> List[ImportActionError]:
+        """
+        Validates the rename action
+        """
+        errors = []
+
+        # Duplicate value validation
+        error = self._validate_value(indexed_actions)
+        if error:
+            errors.append(error)
+
+        return errors
+
+    def execute(self):
+        """
+        Rename a tag
+        """
+        tag = self._get_tag()
+        tag.value = self.tag.value
+        tag.save()
+
+
+class DeleteTag(ImportAction):
+    """
+    Action for delete a Tag
+
+    Action created if the action of the tag is 'delete'
+
+    Does not require validations
+    """
+
+    def __str__(self):
+        taxonomy_tag = self._get_tag()
+        return str(_(f"Delete tag (id={taxonomy_tag.id})"))
+
+    name = "delete"
+
+    @classmethod
+    def valid_for(cls, taxonomy: Taxonomy, tag) -> bool:
+        """
+        Validates if the action is delete and the tag exists
+        """
+        try:
+            taxonomy.tag_set.get(external_id=tag.id)
+            return tag.action == cls.name
```

Comment:
What is the purpose of this `TagItem.action` field? Do we need it?

My impression is that `generate_actions` keeps track of deleted tags and uses `_build_delete_actions` to build the delete actions, without calling `valid_for`.

Reply:
Yes, currently it does not have a major functionality. It arose as an initial idea to have a better management of the actions to be applied, but now it does not make sense. I have removed it here: 8bee009d87dbcf63d1229ee12cfd642fb6ee278a

Reply:
Cool, I think that's cleaner without that field if it's not really required.